server/lib/UserManager.php

```diff
@@ -300,4 +349,33 @@ private function insertOrUpdate($cloudId, $data, $timestamp) {
 
 		return true;
 	}
+
+    /**
+     * Delete all personal data. We keep the basic user entry with the
+     * federated cloud ID in order to propagate the changes
+     *
+     * @param string $cloudId
+     * @return bool
+     */
+    private function deleteDBRecord($cloudId) {
+
+        $stmt = $this->db->prepare('SELECT * FROM users WHERE federationId = :federationId');
```

Comment:
Drop this, just delete to avoid timing problems.
On delete `$stmt->execute();` returns the number of deleted rows.